### PR TITLE
Add some unit test coverage for connection health

### PR DESCRIPTION
**Pull request by csadilek**
## Description

This may seem a bit random :). I wanted to see what it "feels like" to add some unit test coverage to existing components, and picked the first function I came across that seemed interesting. I couldn't convince myself easily that the EWMA calculation works correctly just by looking at the code so I tried to write a test for it. (Spoiler: Yes, it all works correctly)

I chose the path of minimal refactoring and give the tests access to some internals, which I think is fine. The alternative would be to make these internals publicly accessible, which doesn't seem right as it would make the components fragile. The trade-off here is of course that the tests are then more fragile i.e., internal changes of the component can break the tests, but that seems better than exposing those internals via public API. I figured I put this up as a PR, because even in case we don't want to land it in this shape, it's a good discussion to have.

## Checklist
    
- [X] My code follows the style guidelines for this project
- [X] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [X] I have performed a self review of my own code
- [X] I have commented my code PARTICULARLY in hard to understand areas
- [X] I have added thorough tests where needed

### CODE REVIEW

File: src/apps/vpn/connectionhealth.cpp [hunk 1]
```diff
@@ -216,13 +193,17 @@ void ConnectionHealth::dnsPingReceived(quint16 sequence) {
...
   logger.debug() << "Received DNS ping:" << latency << "msec";
+  updateDnsPingLatency(latency);
+}
 
+void ConnectionHealth::updateDnsPingLatency(quint64 latency) {
   if (m_dnsPingInitialized) {
     m_dnsPingLatency *= (PING_BASELINE_EWMA_DIVISOR - 1);
     m_dnsPingLatency += latency;
-    m_dnsPingLatency /= PING_BASELINE_EWMA_DIVISOR;
+    m_dnsPingLatency = round(static_cast<float>(m_dnsPingLatency) / PING_BASELINE_EWMA_DIVISOR);
```
Comment (csadilek) [inline src/apps/vpn/connectionhealth.cpp:206]: This is a bit pendatic on my part. We were just truncating before, happy to reset this.

**Comment by csadilek** [comment]
Looks like the test fails on Windows. 🎯 💯 :) Setting up a Window dev. environment too see what's going on.

**Comment by csadilek** [comment]
Alright, unit tests are 🟢 here now.

**Review by bakulf — APPROVED** [review]

**Review by oskirby — APPROVED** [review]
Looks good! Thank you

**Pull request merged**